### DISCUSSION TITLE
Document processing not working.

### DISCUSSION THREAD

**Discussion by MicahZoltu** [open-webui/open-webui#2848] [General]
# Bug Report

## Description

**Bug Summary:**
When I attach a document to a conversation with `#` and then selecting a document, the AI (Llama 3) responds as though it didn't receive any document.

**Steps to Reproduce:**
1. Add a PDF to Open Web UI
2. Connect to dolphin-llama3 via locally hosted ollama or `meta-llama/Llama-3-70b-chat-hf` via together.xyz (or similar host).
3. Create a new chat and attach the document by typing `#` and then selecting the document from the list.
4. Ask the AI to summarize the document.
5. Notice that it complains about there being no document to summarize

**Expected Behavior:**
The AI can read the document and respond with it as part of its context.

**Actual Behavior:**
Llama 3 70B:
> I apologize, but there is no document provided for me to summarize. Could you please provide the document you would like me to summarize?

dolphin-llama3:
> I'm sorry, but I don't have access to any documents or context information right now. If you can provide me with more details about the topic, I would be happy to help you.

## Environment

- **Open WebUI Version:** v0.2.4
- **Ollama (if applicable):** 0.1.41

- **Operating System:** Windows 11
- **Browser (if applicable):** Firefox

## Reproduction Details

**Confirmation:**

- [X] I have read and followed all the instructions provided in the README.md.
- [X] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [X] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
N/A

**Docker Container Logs:**
```
2024-06-05 21:30:23 {'model': 'dolphin-llama3:latest', 'messages': [{'role': 'user', 'content': "Use the following context as your learned knowledge, inside <context></context> XML tags.\n<context>\n    \n</context>\n\nWhen answer to user:\n- If you don't know, just say that you don't know.\n- If you don't know when you are not sure, ask for clarification.\nAvoid mentioning that you obtained the information from the context.\nAnd answer according to the language of the user's question.\n\nGiven the context information, answer the query.\nQuery: Please summarize this document."}], 'options': {}}
```
```
2024-06-05 21:36:49 {"model": "meta-llama/Llama-3-70b-chat-hf", "stream": true, "messages": [{"role": "user", "content": "Use the following context as your learned knowledge, inside <context></context> XML tags.\n<context>\n    \n</context>\n\nWhen answer to user:\n- If you don't know, just say that you don't know.\n- If you don't know when you are not sure, ask for clarification.\nAvoid mentioning that you obtained the information from the context.\nAnd answer according to the language of the user's question.\n\nGiven the context information, answer the query.\nQuery: Please summarize this document."}]}
```

**Screenshots (if applicable):**
N/A

## Installation Method
Docker

## Additional Information
This used to work, but I didn't update anything and it broke.  Updating images to latest did not fix it.

I tried fiddling with document settings like switching embedding model engine to ollama and resetting vector storage, but this didn't change the symptoms.

**Comment by justinh-rahb** [comment]
> I tried fiddling with document settings like switching embedding model engine to ollama and resetting vector storage, but this didn't change the symptoms.

This is likely why it is not working. Reset your vector database, and try uploading documents again. It's important to remember that changing the embedding model basically makes the existing vectors useless, so the database must always be reset.

**Comment by MicahZoltu** [comment]
It wasn't working *before* I fiddled with the settings.  I fiddled with settings, reset vector DB a couple of times, etc. and nothing fixed it.  Each time I did this I re-added the document, as the UI indicated that this was necessary.

**Reply by justinh-rahb** [reply]
Alright... and the document.. have you only been trying the same one or have you tried others? Did you try manually converting the contents of the PDF to `.txt` to rule out PDF formatting issues? Can you share the specific file in question? Can you get us some actual full logs from the container and your browser? There's a lot of unknowns here, the more things you can tell us that you tried, the better your chances of getting a resolution.

**Reply by MicahZoltu** [reply]
I have tried multiple documents, including one I successfully used previously.

I just reset everything back to defaults in an attempt to capture more logs and test with a text file, and now everything is working.  I have no clue what caused it to break previously as I didn't touch any of the document settings until *after* it wasn't working, but whatever the problem it seems to be resolved now.

If anyone else runs into this problem, I did the following:
1. Deleted all documents.
2. Reset document settings back to default embedding model engine.
3. Saved settings.
4. Reset vector storage.
5. Add the document back.
6. Attempt to query the model about the document.

**Reply by PierreSchwang** [reply]
Hi! I tried your steps but sadly fail to make it run either way (it did not work in first place either). I'm using all the defaults - even with Ollama set as the embedding provider - it's not working. I can see in the Open-WebUI log that the document was read and stored in chomadb - but it seems like it's not sent along to Ollama?

As far as I can tell in the developer tools, the collection id and document title is sent along, e.g. the call to the `/ollama/api/chat` rest call:
```json
{
   "model":"llama3:latest",
   "messages":[
      {
         "role":"user",
         "content":"summarize the document"
      }
   ],
   "options":{
      
   },
   "docs":[
      {
         "type":"doc",
         "collection_name":"f011397ba4e843785eca817dc38e993cda821a8ba4bf3490dda3d83295d8efd",
         "name":"censoreddocx",
         "title":"censored.docx",
         "filename":"censored.docx",
         "content":{
            
         },
         "user_id":"aad8bfcb-fa3e-4732-a5fc-0ee2d4ee6ea3",
         "timestamp":1717683467,
         "upload_status":true
      }
   ],
   "citations":true,
   "chat_id":"3e13563e-a047-465b-a246-cc7a1e3d27db"
}
```

The answer just indicates, that it has no idea what document I'm talking about or tries to talk about some random document out there in the wild. I don't really know how the collection name helps the ollama backend seemingly no direct access to the chromadb itself - maybe that's the issue or it's just me not fully understanding the architecture. The content object is empty, so that is most likely the reason.

But even with debug logging enabled, i can't really find anything

**Reply by PierreSchwang** [reply]
Regarding the original discussion post, I can't even find any log output remotely similar to the posted ones (`2024-06-05 21:30:23 {'model': 'dolphin-llama3:latest', 'messages': [{...`) - shouldn't that be logged? Is that because of some kind of misconfiguration?

**Reply by Jaakuice** [reply]
Could this be query params top k being too large?

**Reply by PierreSchwang** [reply]
Doesn't seem to change anything for me sadly, it was set to default and I've cranked it down to 5 without any change - except llama3 starting to get salty lol
> I think we're stuck in a loop!
>
>As I've said multiple times, there is NO DOCUMENT provided for me to summarize.
>
>If you'd like to share a text or document with me, I'll be happy to help you summarize it. Just let me know what you have!

**Comment by drupol** [comment]
Have you tried to upload the document in the chat instead of using an existing document from the workspace? It works for me. See the issue I opened: https://github.com/open-webui/open-webui/discussions/2906

**Reply by PierreSchwang** [reply]
In my case, I do get a response either way - both being without any context though to be fair.

**Reply by drupol** [reply]
Indeed, it's a different issue.

**Comment by petunder** [comment]
I faced the same problem as you. The solution is as follows (the order is important):

1. Remove all documents.
2. Reset the Vector Storage.
3. Load the documents again.
4. Use the newly loaded documents in a new chat (the old ones won't work).
